server/stats.go

```diff
@@ -6,18 +6,56 @@
 package main
 
 import (
+	"encoding/json"
 	"expvar"
 	"log"
 	"net/http"
 	"runtime"
 	"time"
 )
 
+// A simple implementation of histogram expvar.Var.
+// `Bounds` specifies the histogram buckets as follows (length = len(bounds)):
+//     (-inf, Bounds[i]) for i = 0
+//     [Bounds[i-1], Bounds[i]) for 0 < i < length
+//     [Bounds[i-1], +inf) for i = length
+type histogram struct {
+	Count          int64     `json:"count"`
+	CountPerBucket []int64   `json:"count_per_bucket"`
+	Bounds         []float64 `json:"bounds"`
+}
+
+func (h *histogram) addSample(v float64) {
+	h.Count++
+	var count *int64
+	var i int
+	var b float64
+	for i, b = range h.Bounds {
```

Comment:
There are a lot of boundaries, they are sorted, and this method is called frequently. Please consider using `sort.Search`.

Reply:
Sure. It was supposed to be too generic. Of course, binary search works better here.